slash/resuming.py

```diff
@@ -1,79 +1,95 @@
-import itertools
-import json
 import os
-
 import logbook
-
-from .utils.path import ensure_directory
-
-_logger = logbook.Logger(__name__)
+from sqlalchemy import Column, DateTime, String, Integer
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+from contextlib import contextmanager
+from datetime import datetime
+from slash.plugins import manager
 
 _RESUME_DIR = os.path.expanduser("~/.slash/session_states")
-_RESUME_COUNTER = itertools.count()
-_MAX_NUM_SAVED_SESSIONS = 10
-
-LATEST = object()
-
-
-def save_resume_state(session_result):
-    resume_filename = _generate_resume_filename(session_result.session.id)
-    tmp_filename = resume_filename + ".tmp"
-    ensure_directory(os.path.dirname(tmp_filename))
-    with open(tmp_filename, "w") as f:
-        json.dump({
-            "tests": [
-                {"address": str(result.test_metadata.address), "needs_rerun":
-                 result.is_failure() or result.is_error() or not result.is_started()}
-                for result in session_result.iter_test_results()
-            ],
-        }, f)
-    os.rename(tmp_filename, resume_filename)
-    _logger.debug('Saved resume state to {0}', resume_filename)
-    _cleanup_old_files()
-
+_DB_NAME = 'resume_state.db'
+_logger = logbook.Logger(__name__)
+Base = declarative_base()
+session = sessionmaker()
+is_db_initialized = False
+
+class ResumeState(Base):
+    __tablename__ = 'resume_state'
+    id = Column(Integer, primary_key=True)
+    session_id = Column(String, nullable=False, index=True)
+    test_name = Column(String, nullable=False)
+
+class SessionMetadata(Base):
+    __tablename__ = 'session_metadata'
+    session_id = Column(String, primary_key=True)
+    src_folder = Column(String, nullable=False, index=True)
+    created_at = Column(DateTime, nullable=False, index=True)
+
+def init_db():
+    engine = create_engine('sqlite:///{0}/{1}'.format(_RESUME_DIR, _DB_NAME))
+    session.configure(bind=engine)
+    Base.metadata.create_all(engine)
+
+@contextmanager
+def connecting_to_db():
+    global is_db_initialized  # pylint: disable=global-statement
```

Comment:
Call it `_db_initiatlized`. Otherwise:
1. It looks like a function
2. It looks public